strawberry/schema/schema_converter.py

```diff
@@ -163,6 +175,38 @@ def from_input_field(self, field: StrawberryField) -> GraphQLInputField:
             deprecation_reason=field.deprecation_reason,
         )
 
+    FieldType = TypeVar("FieldType", GraphQLField, GraphQLInputField)
+
+    @staticmethod
+    def _get_thunk_mapping(
+        fields: List[StrawberryField],
+        name_converter: Callable[[StrawberryField], str],
+        field_converter: Callable[[StrawberryField], FieldType],
```

Comment:
Is there a reason these pull from args instead of using `self.` variables directly (which is where the values come from anywhere). Is this abstraction beneficial?

Reply:
`name_converter` can be pulled from `self` directly since all type definitions seem to use the same converter, but from what I've seen `field_converter` changes based on if it is called from `self.from_input_object` or `self.from_object`. The reason I made an abstraction here is to separate the logic of the filtering and the creation of a GraphQL Core  `ThunkMapping` from that of selecting which converter(s) to use for the field. This is to prevent having to change `_get_thunk_mapping` if new types are added. With this in mind I didn't pull the converter(s) from `self` directly so that other functions (or in the future a mapping of Strawberry types to GraphQL types ) can decide which one to use.

The amount of code here could be reduced further if there is a way to have a single `get_graphql_fields` method that is responsible for selecting the right converters. But right now that's difficult to do since the static type of `type_definition` in `from_input_object` evaluates to `Any`. I couldn't find a way to address that quickly so I left it as is for now.